src/test/rmc-dependency-test/diamond-dependency/run-dependency-test.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+# Test for platform
+platform=`uname -sp`
+if [[ $platform != "Linux x86_64" ]]; then
+  echo "Codegen script only works on Linux x86 platform"
+  exit 0
+fi
+
+# Log output
+LOG="/tmp/dependency_test_log.txt"
+
+# Compile crates with RMC backend
+cd $(dirname $0)
+rm -rf build
+CARGO_TARGET_DIR=build RUST_BACKTRACE=1 RUSTFLAGS="-Z codegen-backend=gotoc --cfg=rmc" RUSTC=rmc-rustc cargo build --target x86_64-unknown-linux-gnu 2> $LOG
```

Comment:
What happens if we use the osx target here and build on mac

Reply:
IIRC it requires rebuilding the std lib on mac, which then fails. We can look at this again together.